### PR TITLE
Fix for illogical Obsidian and additional tiles redo

### LINKED ISSUE #219: [FEATURE] Adding raw obsidian for logic reasons and corresponding Decor Tiles
**Additional Info (12/07/2021):**
Add DecorTiles for more tile diversity. 

**Is your feature request related to a problem? Please describe.**
The way water and lava make obsidian “planks” makes no sense and is illogical.

**Describe the solution you'd like**
I recommend changing the texture and name to something like “Raw Obsidian” can be a new tile which in this case can be mined and used to make Obsidian Planks.

**Describe alternatives you've considered**
An alternative that requires less coding is renaming it to obsidian and changing the texture.

### PR DESCRIPTION
**Pull request by Litorom**
Fix for illogical Obsidian and additional tiles such as Stone (Raw), Ornate Stone, Obisidian (Raw), and Obsidian Tiles.
Fixes #219.

### CODE REVIEW

File: src/main/java/minicraft/item/TileItem.java [hunk 1]
```diff
@@ -28,10 +28,14 @@
...
 		items.add(new TileItem("Plank Wall", (new Sprite(1, 5, 0)), "Wood Wall", "Wood Planks"));
 		items.add(new TileItem("Wood Door", (new Sprite(2, 5, 0)), "Wood Door", "Wood Planks"));
+		items.add(new TileItem("Stone", (new Sprite(2, 0, 0)), "Stone", "hole", "water", "cloud", "lava"));
 		items.add(new TileItem("Stone Brick", (new Sprite(3, 5, 0)), "Stone Bricks", "hole", "water", "cloud", "lava"));
+		items.add(new TileItem("Ornate Stone", (new Sprite(3, 5, 0)), "Ornate Stone", "hole", "water", "cloud", "lava"));
 		items.add(new TileItem("Stone Wall", (new Sprite(4, 5, 0)), "Stone Wall", "Stone Bricks"));
 		items.add(new TileItem("Stone Door", (new Sprite(5, 5, 0)), "Stone Door", "Stone Bricks"));
+		items.add(new TileItem("Raw Obsidian", (new Sprite(9, 5, 0)), "Raw Obsidian", "hole", "water", "cloud", "lava"));
 		items.add(new TileItem("Obsidian Brick", (new Sprite(6, 5, 0)), "Obsidian", "hole", "water", "cloud", "lava"));
+		items.add(new TileItem("Obsidian Tiles", (new Sprite(6, 5, 0)), "Obsidian Tiles","hole", "water", "cloud", "lava"));
```
Comment (Makkkkus) [inline src/main/java/minicraft/item/TileItem.java:38]: Rename obsidian tiles to ornate obsidian

File: src/main/java/minicraft/level/tile/DecorTile.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import minicraft.item.Items;
+import minicraft.item.ToolItem;
+import minicraft.level.Level;
+
+public class DecorTile extends Tile {
+	protected Material type;
+	private Sprite sprite;
+
+	protected DecorTile(Material type) {
+		super((type == Material.Obsidian ? "Obsidian Tiles" : type == Material.Stone ? "Stone" :type.name() + " Bricks"), (Sprite) null);
```
Comment (Makkkkus) [inline src/main/java/minicraft/level/tile/DecorTile.java:18]: change [type.name() + " Bricks"] to ["Decorated " + type.name()] for redundancy

**Review by Makkkkus — CHANGES_REQUESTED** [review]
Very good, but some changes have to be addressed before merging.

File: src/main/java/minicraft/level/tile/MaterialTile.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import minicraft.item.Items;
+import minicraft.item.ToolItem;
+import minicraft.level.Level;
+
+public class MaterialTile extends Tile {
+	protected Material type;
+	private Sprite sprite;
+
+	protected MaterialTile(Material type) {
+		super((type == Material.Stone ? "Stone" : type == Material.Obsidian ? "Raw Obsidian" :type.name() + " Bricks"), (Sprite) null);
```
[the remark this answers is not in the document]
Comment (Litorom) [reply src/main/java/minicraft/level/tile/MaterialTile.java:18]: For some reason I can’t, it shows an error. I’ll look more into why later

File: src/main/java/minicraft/item/Recipes.java [hunk 4]
```diff
@@ -22,8 +22,11 @@
 		workbenchRecipes.add(new Recipe("Torch_2", "Wood_1", "coal_1"));
 		workbenchRecipes.add(new Recipe("Lantern_1", "Wood_8", "slime_4", "glass_3"));
 		workbenchRecipes.add(new Recipe("Stone Brick_2", "Stone_2"));
+		workbenchRecipes.add(new Recipe("Ornate Stone_2", "Stone_2"));
 		workbenchRecipes.add(new Recipe("Stone Wall_1", "Stone Brick_3"));
 		workbenchRecipes.add(new Recipe("Stone Door_1", "Stone Brick_5"));
+		workbenchRecipes.add(new Recipe("Obsidian Brick_2", "Raw Obsidian_2"));
+		workbenchRecipes.add(new Recipe("Obsidian Tiles_2", "Raw Obsidian_2"));
```
[the remark this answers is not in the document]
Comment (Litorom) [reply src/main/java/minicraft/item/Recipes.java:29]: Well that’s the original recipe, but I can do that.

**Review by Makkkkus — APPROVED** [review]
Good

**Pull request merged**